plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -648,4 +662,74 @@ protected TestView createSleepingView(Duration minimalSleepDuration)
     {
         throw new UnsupportedOperationException();
     }
+
+    @Test
+    public void testMetadataDeleteWithBigintEquality()
+    {
+        if (!hasBehavior(SUPPORTS_METADATA_DELETE)) {
+            assertQueryFails("DELETE FROM region WHERE regionkey = 1", "This connector does not support deletes");
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM region", 5);
+        assertUpdate("DELETE FROM " + tableName + " WHERE regionkey = 1", 1);
+        assertQuery("SELECT count(*) FROM " + tableName, "VALUES 4");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMetadataDeleteWithVarcharEquality()
+    {
+        if (!hasBehavior(SUPPORTS_METADATA_DELETE) || !hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_EQUALITY)) {
+            assertQueryFails(
+                    "DELETE FROM nation WHERE name = 'ALBANIA'",
+                    "(.*[Dd]elet(e|ing).*(not |un)supported.*)" +
+                            "|(This connector does not support deletes)" +
+                            "|(Unsupported delete)");
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(col) AS VALUES 'a', 'A', null", 3);
+        assertUpdate("DELETE FROM " + tableName + " WHERE col = 'A'", 1);
```

Comment:
isn't `col = 'A'` equality belong to `SUPPORTS_ROW_LEVEL_DELETE` ? 
This could be metadata condition for some kind of Skip Indexes (e.g. https://github.com/citusdata/cstore_fdw#using-skip-indexes) but i'm not sure this is a `SUPPORTS_METADATA_DELETE` by default.